### PR TITLE
Modernizing Practice Settings page

### PR DESCRIPTION
**Pull request by robertdown**
Quick little update to the Practice Settings page

I get easily distracted when doing things like adding a pharmacy for testing and start updating the UI. This is just about ready to go, just.... where is the tab title defined? Obviously it's not correct. Just gotta fix that and this should be good.

It's super straight forward, just added the sidebar and added some classes to the tables

### CODE REVIEW

**Comment by robertdown** [comment]
Couple of screenshots

screen shot 2017-04-11 at 00 16 44

screen shot 2017-04-11 at 00 16 47

File: templates/practice_settings/general_list.html [hunk 1]
```diff
@@ -3,33 +3,58 @@
 {php}html_header_show();{/php}
 
 <link rel="stylesheet" href="{$CSS_HEADER}" type="text/css">
+<link rel="stylesheet" type="text/css" href="{php} echo $GLOBALS['assets_static_relative']{/php}/bootstrap-3-3-4/dist/css/bootstrap.css">
```
Comment (bradymiller) [inline templates/practice_settings/general_list.html]: add the bootstrap rtl library if applicable (like done in the demographics)

File: templates/practice_settings/general_list.html [hunk 2]
```diff
@@ -3,33 +3,58 @@
...
 
-<div>
-    <b>{xl t='Practice Settings'}</b>
-</div>
+<nav class="navbar navbar-default navbar-fixed-top">
+    <div class="container-fluid">
+        <!-- Brand and toggle get grouped for better mobile display -->
+        <div class="navbar-header">
+            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
+                <span class="sr-only">Toggle navigation</span>
```
Comment (bradymiller) [inline templates/practice_settings/general_list.html]: translate

File: templates/practice_settings/general_list.html [hunk 3]
```diff
@@ -3,33 +3,58 @@
...
-    <br/>
-    <div class="section-header">
-        <b>{$ACTION_NAME}</b>
-    </div>
-    <div class="tabContainer">
-        <div class="tab current" style="width:100%">
-            {$display}
+<div class="container-fluid">
+    <div class="row">
+        <div class="col-xs-8 col-sm-2 col-md-2 sidebar sidebar-left sidebar-md-show">
```
Comment (bradymiller) [inline templates/practice_settings/general_list.html]: make it right when RTL language is being used

Comment (bradymiller) [inline templates/practice_settings/general_list.html]: Getting weird thing happening:
if minimimize screen, then the menu goes away and no menu icon shows up
if continue to minimize the screen, then the menu icon shows up at top right, but it doesn't work (getting a bunch of  js errors stating jquery 1.9.1 at least is needed, so guessing that is causing issues; I would bump it to jquery 3.1.1).

Comment (bradymiller) [inline templates/practice_settings/general_list.html]: oh man, I just noted fancybox is also included. Is that actually being used anywhere in this code?

File: templates/practice_settings/general_list.html [hunk 4]
```diff
@@ -3,33 +3,58 @@
...
+    <div class="container-fluid">
+        <!-- Brand and toggle get grouped for better mobile display -->
+        <div class="navbar-header">
+            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
+                <span class="sr-only">Toggle navigation</span>
+                <span class="icon-bar"></span>
+                <span class="icon-bar"></span>
+                <span class="icon-bar"></span>
+            </button>
+            <a class="navbar-brand" href="#">{php}echo xlt('Practice Settings'){/php}</a>
```
Comment (bradymiller) [inline templates/practice_settings/general_list.html]: To see how the tab title is selected(it's a bit hokey :)), see here:
https://github.com/openemr/openemr/blob/master/interface/main/tabs/js/custom_bindings.js#L32
(in this case I think it was grabbing the first bolded term)

If set the title to be translated Practice Settings in the head section, I think you will be good.

**Comment by bradymiller** [comment]
On a good note, there were not surprises in frames mode.

**Comment by bradymiller** [comment]
Couldn't find fancybox being used anywhere in the practice settings, so should be able to yank that include and go to jquery 3.1.1 .

**Comment by bradymiller** [comment]
Overall looks nice!
Just some minor things to work out.

File: templates/practice_settings/general_list.html (hunk 1 above)
Comment (robertdown) [inline templates/practice_settings/general_list.html]: Will add

File: templates/practice_settings/general_list.html (hunk 2 above)
Comment (robertdown) [inline templates/practice_settings/general_list.html]: Will translate

File: templates/practice_settings/general_list.html (hunk 3 above)
Comment (robertdown) [inline templates/practice_settings/general_list.html]: Will add RTL support, remove fancybox, and bump to jQuery 3.1.1

File: templates/practice_settings/general_list.html (hunk 4 above)
Comment (robertdown) [inline templates/practice_settings/general_list.html]: Wow.... that's an interesting solution to getting the title.... 👎

File: templates/practice_settings/general_list.html (hunk 2 above)
Comment (robertdown) [inline templates/practice_settings/general_list.html]: Done

File: templates/practice_settings/general_list.html (hunk 3 above)
Comment (robertdown) [inline templates/practice_settings/general_list.html]: LTR done

File: templates/practice_settings/general_list.html (hunk 4 above)
Comment (robertdown) [inline templates/practice_settings/general_list.html]: Fixed by adding `<title>` element

**Comment by robertdown** [comment]
@bradymiller take a look, think I addressed all the concerns. Think being the operative word there

File: templates/practice_settings/general_list.html [hunk 5]
```diff
@@ -1,35 +1,69 @@
...
+    <link rel="stylesheet" type="text/css"
+          href="{php} echo $GLOBALS['assets_static_relative']{/php}/bootstrap-3-3-4/dist/css/bootstrap.css">
+    {if $direction == 'right'}
+    <link rel="stylesheet" type="text/css"
+          href="{php} echo $GLOBALS['assets_static_relative']{/php}/bootstrap-rtl-3-3-4/dist/css/bootstrap-rtl.css">
+    {/if}
+    <link rel="stylesheet" type="text/css"
+          href="{php} echo $GLOBALS['assets_static_relative']{/php}/bootstrap-sidebar-0-2-2/dist/css/sidebar.css">
+    <script type="text/javascript"
+            src="{php}echo $GLOBALS['assets_static_relative']{/php}/jquery-min-3-3-1/index.js"></script>
```
Comment (bradymiller) [inline templates/practice_settings/general_list.html]: I think I gave you bad advice. Was supposed to be jquery version 3-1-1 .

**Comment by bradymiller** [comment]
Looking much better.
For the testing I changed the jquery to 3-1-1 and removed browser cache/cookies.
Noting the following issues on testing:
1. The sidebar links have odd link behavior (where if visit a link it's a different color which seems odd).
2. When testing in Hebrew which is an rtl language, the sidebar menu is not going to right (stays on the left); I kind of wonder if bootstrap rtl does this already and you may be reverting it by placing direction there.
3. Still getting the following odd behavior:
When browser width is wide, sidebar menu shows up correctly.
If decrease browser width, then the menu goes away and no menu icon shows up.
If then continued ti decrease browser width, then the menu icon shows up at top right, but it doesn't work when click on it.

**Comment by robertdown** [comment]
You didn't give me bad advice on jquery, I just can't read... was trying to get the updates in before calling it a night and I rushed it. I have an idea as to why you're getting odd behavior. Also, I think you may be right re: the RTL issue. Was tinkering with it last night. Will keep looking

**Comment by robertdown** [comment]
**The sidebar links have odd link behavior (where if visit a link it's a different color which seems odd).**
Styling issue, please don't let this be a blocker for this PR. Will resolve in separate PR

**When testing in Hebrew which is an rtl language, the sidebar menu is not going to right**
Should now be resolved

**Still getting the following odd behavior: When browser width is wide, sidebar menu shows up correctly.**
Now resolved. Button should only display when width is very small and should toggle the menu.

Note I'm not happy with the look of the toggle button, will fix in a separate PR (Along with the weird link things)

@bradymiller ready for (hopefully the last review)

**Comment by bradymiller** [comment]
@robertdown ,
Agree the link behavior and toggle button are minor style issues that can be addressed later.

Did heavy testing on varied themes with tabs/frames, and there is only 1 more issue that needs to be addressed before bringing this in. On the tan and other legacy themes, getting weird behavior when click the menu icon (guessing just need to place the .sidebar background in the other theme files to fix this):

Here's wide screen that looks good:
screenshot_2017-04-14_01-23-29

Here's narrow screen that looks good:
screenshot_2017-04-14_01-23-54

However, note what happens when click on the menu icon:
screenshot_2017-04-14_01-24-04

After you address this issue, then this is ready to come into the codebase. When you bring this into the codebase, definitely use github's 'Squash and merge' selection which will rebase all the commits into 1 to make it much easier to debug if issues come up in the future related to these changes.

Also, I think in the future, the automated code standard changes may be better to do in a separate PR after (of course, this depends on how extensive the code changes are) since this can bury the "real" code changes which make the code review more difficult.

thanks! -brady

**Pull request merged**